src/main/java/com/intel/genomicsdb/spark/GenomicsDBInputFormat.java

```diff
@@ -65,17 +76,173 @@
+    String amendedQuery = "{\n";
+    amendedQuery += indentString + "\"workspace\": \""+inputSplit.getPartitionInfo().getWorkspace()+"\",\n";
+    amendedQuery += indentString + "\"array\": \""+inputSplit.getPartitionInfo().getArrayName()+"\",\n";
+    if (inputSplit.getQueryInfo().getBeginPosition() == inputSplit.getQueryInfo().getEndPosition()) {
+      amendedQuery += indentString + "\"query_column_ranges\": [["+inputSplit.getQueryInfo().getBeginPosition()+"]]";
+    }
+    else {
+      amendedQuery += indentString + "\"query_column_ranges\": [[["+inputSplit.getQueryInfo().getBeginPosition()+","
+	      +inputSplit.getQueryInfo().getEndPosition()+"]]]";
+    }
+
+    try {
+
+      JSONParser parser = new JSONParser();
+      JSONObject obj = (JSONObject)parser.parse(new FileReader(queryJson));
+  
+      if (obj.containsKey("query_row_ranges")) {
+        amendedQuery += ",\n" + indentString + "\"query_row_ranges\": "+obj.get("query_row_ranges").toString()+"";
+      }
+      if (obj.containsKey("vid_mapping_file")) {
+        amendedQuery += ",\n" + indentString + "\"vid_mapping_file\": \""+obj.get("vid_mapping_file").toString()+"\"";
+      }
+      if (obj.containsKey("callset_mapping_file")) {
+        amendedQuery += ",\n" + indentString + "\"callset_mapping_file\": \""+obj.get("callset_mapping_file").toString()+"\"";
+      }
+      if (obj.containsKey("vcf_header_filename")) {
+        amendedQuery += ",\n" + indentString + "\"vcf_header_filename\": "+obj.get("vcf_header_filename").toString();
+      }
+      if (obj.containsKey("query_attributes")) {
+        amendedQuery += ",\n" + indentString + "\"query_attributes\": "+obj.get("query_attributes").toString();
+      }
+      if (obj.containsKey("reference_genome")) {
+        amendedQuery += ",\n" + indentString + "\"reference_genome\": \""+obj.get("reference_genome").toString()+"\"";
+      }
+      if (obj.containsKey("segment_size")) {
+        amendedQuery += ",\n" + indentString + "\"segment_size\": "+obj.get("segment_size").toString();
+      }
+      if (obj.containsKey("produce_GT_field")) {
+        amendedQuery += ",\n" + indentString + "\"produce_GT_field\": "+obj.get("produce_GT_field").toString();
+      }
+      if (obj.containsKey("produce_FILTER_field")) {
+        amendedQuery += ",\n" + indentString + "\"produce_FILTER_field\": "+obj.get("produce_FILTER_field").toString();
+      }
+      amendedQuery += "\n}\n";
+    }
+    catch (FileNotFoundException e) {
+      e.printStackTrace();
+      return null;
+    }
+    catch (IOException e) {
+      e.printStackTrace();
+      return null;
+    }
+    catch (ParseException e) {
+      e.printStackTrace();
+      return null;
+    }
+    
+    File tmpQueryFile = File.createTempFile("queryJson", ".json");
+    tmpQueryFile.deleteOnExit();
+    FileWriter fptr = new FileWriter(tmpQueryFile);
+    fptr.write(amendedQuery);
+    fptr.close();
+    return tmpQueryFile.getAbsolutePath();
+  }
+
```

Comment:
Ok for now since you are managing the Spark interface - but it's probably easier to use the Protobuf API in the future

Reply:
Agreed. We are planning to move to using protobuf in the near future. Thanks.